### PR TITLE
Remove cache-path from project config (hover.yaml)

### PR DESCRIPTION
**Pull request by GeertJohan**
I also added a bunch of TODO comments on things that I noticed while working on this and would like to improve.

### CODE REVIEW

File: internal/enginecache/path.go [hunk 1]
```diff
@@ -2,31 +2,17 @@ package enginecache
...
 	if err != nil {
-		log.Errorf("Failed to resolve home path: %v", err)
-		os.Exit(1)
+		log.Warnf("Failed to resolve cache path: %v", err)
 	}
-
-	var p string
-	switch runtime.GOOS {
-	case "linux":
-		p = filepath.Join(homePath, ".cache")
```
Comment (james-lawrence) [inline internal/enginecache/path.go:22]: I assume os.UserCacheDir fixes all those other paths?

Comment (GeertJohan) [reply internal/enginecache/path.go:22]: Yes, in newer versions of Go we have UserCacheDir available and it basically does what we did in our code, but also takes into account other standards such as `$XDG_CONFIG_HOME`.

**Comment by GeertJohan** [comment]
Close was misclick.

**Comment by GeertJohan** [comment]
@james-lawrence 
> the default cache path seems to have lost all the suffixes and they're not restored anywhere?

I do not understand. To which suffixes do you refer?

> all these global variables with prefix names makes it hard to keep track.

I agree, it's gotten a bit out of hand and could use some refactoring. I'm going to make a plan for that.

**Comment by james-lawrence** [comment]
> I do not understand. To which suffixes do you refer?

i was just referring to the user.CacheDir stuff. we've already covered it.

**Pull request merged**